core/tauri/src/menu/check.rs

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019-2023 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use crate::{run_main_thread, AppHandle, Manager, Runtime};
+
+/// A menu item inside a [`Menu`] or [`Submenu`] and contains only text.
+///
+/// [`Menu`]: super::Menu
+/// [`Submenu`]: super::Submenu
+pub struct CheckMenuItem<R: Runtime> {
+  pub(crate) id: u32,
+  pub(crate) inner: muda::CheckMenuItem,
+  pub(crate) app_handle: AppHandle<R>,
+}
+
+impl<R: Runtime> Clone for CheckMenuItem<R> {
+  fn clone(&self) -> Self {
+    Self {
+      id: self.id,
+      inner: self.inner.clone(),
+      app_handle: self.app_handle.clone(),
+    }
+  }
+}
+
+/// # Safety
+///
+/// We make sure it always runs on the main thread.
+unsafe impl<R: Runtime> Sync for CheckMenuItem<R> {}
+unsafe impl<R: Runtime> Send for CheckMenuItem<R> {}
+
+impl<R: Runtime> super::sealed::IsMenuItemBase for CheckMenuItem<R> {
+  fn inner(&self) -> &dyn muda::IsMenuItem {
+    &self.inner
+  }
+}
+
+impl<R: Runtime> super::IsMenuItem<R> for CheckMenuItem<R> {
+  fn kind(&self) -> super::MenuItemKind<R> {
+    super::MenuItemKind::Check(self.clone())
+  }
+
+  fn id(&self) -> u32 {
+    self.id
+  }
+}
+
+impl<R: Runtime> CheckMenuItem<R> {
+  /// Create a new menu item.
+  ///
+  /// - `text` could optionally contain an `&` before a character to assign this character as the mnemonic
+  /// for this menu item. To display a `&` without assigning a mnemenonic, use `&&`.
+  pub fn new<M: Manager<R>, S: AsRef<str>>(
+    manager: &M,
+    text: S,
+    enabled: bool,
+    checked: bool,
+    acccelerator: Option<S>,
+  ) -> Self {
+    let item = muda::CheckMenuItem::new(
+      text,
+      enabled,
+      checked,
+      acccelerator.and_then(|s| s.as_ref().parse().ok()),
+    );
+    Self {
+      id: item.id(),
+      inner: item,
+      app_handle: manager.app_handle(),
+    }
+  }
+
+  /// The application handle associated with this type.
+  pub fn app_handle(&self) -> AppHandle<R> {
```

Comment:
I don't think we need to clone here

Reply:
This was done to just match Manager::app_handle implementation, should I keep it or change `Manager::app_handle` to also return a reference ?

Reply:
I think with muda taking references, it would be convenient to return references everywhere instead (also on app.handle).

Reply:
not sure if this is what you wanted but here goes https://github.com/tauri-apps/tauri/pull/7535/commits/84467d46cae9e54d4a659d998b1070ddf69ed004